packages/hop-node/src/watchers/BondTransferRootWatcher.ts

```diff
@@ -151,72 +121,34 @@ class BondTransferRootWatcher extends BaseWatcher {
     )
     logger.debug('committedAt:', committedAt)
     logger.debug('destinationChainId:', destinationChainId)
+    logger.debug('sourceChainId:', sourceChainId)
     logger.debug('transferRootHash:', transferRootHash)
     logger.debug('transferRootId:', transferRootId)
     logger.debug('totalAmount:', this.bridge.formatUnits(totalAmount))
     logger.debug('transferRootId:', transferRootId)
 
     await this.waitTimeout(transferRootHash, totalAmount)
-    dbTransferRoot = await this.db.transferRoots.getByTransferRootHash(
-      transferRootHash
-    )
-    if (!dbTransferRoot) {
-      logger.warn('no transfer root')
-      return
-    }
 
-    logger.debug(
-      'dbTransferRoot transferRootHash:',
-      dbTransferRoot.transferRootHash
-    )
-    logger.debug(
-      'dbTransferRoot totalAmount:',
-      this.bridge.formatUnits(dbTransferRoot.totalAmount)
-    )
-    logger.debug(
-      'dbTransferRoot destinationChainId:',
-      dbTransferRoot.destinationChainId
-    )
-    logger.debug('dbTransferRoot sourceChainId:', sourceChainId)
-    logger.debug('dbTransferRoot committedAt:', dbTransferRoot.committedAt)
-    logger.debug('dbTransferRoot committed:', dbTransferRoot.committed)
-    logger.debug('dbTransferRoot sentBondTx:', !!dbTransferRoot.sentBondTx)
-    const pendingTransfers: string[] = Object.values(
-      dbTransferRoot.transferIds || []
-    )
+    const pendingTransfers: string[] = transferIds || []
     logger.debug('transferRootHash transferIds:', pendingTransfers)
     if (pendingTransfers.length) {
       const tree = new MerkleTree(pendingTransfers)
       const rootHash = tree.getHexRoot()
       logger.debug('calculated transfer root hash:', rootHash)
       if (rootHash !== transferRootHash) {
-        logger.warn('calculated transfer root hash does not match')
-      }
-    }
-
-    dbTransferRoot = await this.db.transferRoots.getByTransferRootHash(
-      transferRootHash
-    )
-    if (
-      (dbTransferRoot?.sentBondTx || dbTransferRoot?.bonded) &&
-      dbTransferRoot.sentBondTxAt
-    ) {
-      // skip if a transaction was sent in the last 10 minutes
-      if (dbTransferRoot.sentBondTxAt + TX_RETRY_DELAY_MS > Date.now()) {
-        logger.debug(
-          transferRootHash,
-          'sent?:',
-          !!dbTransferRoot.sentBondTx,
-          'bonded?:',
-          !!dbTransferRoot?.bonded
-        )
+        logger.error('calculated transfer root hash does not match')
       }
       return
     }
 
-    const hasPositiveBalance = await l1Bridge.hasPositiveBalance()
-    if (!hasPositiveBalance) {
-      logger.warn('bonder requires positive balance to bond transfer root')
+    const availableCredit = await l1Bridge.getAvailableCredit()
+    const bondAmount = await l1Bridge.getBondForTransferAmount(totalAmount)
+    if (availableCredit.lt(bondAmount)) {
+      logger.warn(
```

Comment:
would adding a slack notification be good to have here as well?

Reply:
Fixed in d88e3e18b3c41f94bc0ca094df6864013e7eb9bf.